pkg/platform/controller/cluster/cluster_controller.go

```diff
@@ -131,7 +132,17 @@ func (c *Controller) addCluster(obj interface{}) {
 func (c *Controller) updateCluster(old, obj interface{}) {
 	oldCluster := old.(*platformv1.Cluster)
 	cluster := obj.(*platformv1.Cluster)
-	if !c.needsUpdate(oldCluster, cluster) {
+
+	controllerNeedUpddateResult := c.needsUpdate(oldCluster, cluster)
+	var providerNeedUpddateResult bool
+
+	provider, _ := clusterprovider.GetProvider(cluster.Spec.Type)
+	if provider == nil {
+		c.log.Infof("get cluster provider with %s failed", cluster.Spec.Type)
```

Comment:
modify log to error level. see https://github.com/tkestack/tke/pull/1503/files#diff-303134097085051ccecc6547b55796f466ec830f43c56fcb19736bd3ee2bb35fL304